File: 1 - BackEnd/Vanilla/PHP/estructuras_de_control/try_catch.php
<?php

/* ##########==========================================########## */
/* ######===--- Try Catch y lanzar excepciones throw ---===###### */
/* ##########==========================================########## */

// ------------------------------ //
// ------ Excepcion normal ------ //
// ------------------------------ //

try{

    // La palabra reservada (throw) permite lanzar excepciones.
	throw "Variable: $hola";

// Aqui se atrapa el error.
}catch(Throwable $error){
    // El error se recibe.

    printf("%s", $error);
}finally{
    // Bloque final de ejecucion.
}

// -------------------------------- //
// ------ Objeto (Exception) ------ //
// -------------------------------- //

try
{
    // Se puede crear una instancia (Exception) y lanzarla.
    throw new Exception("Mensaje");
}
catch(Exception $error){
    // El objeto lanzado posee el metodo (getMessage) que devuelve el mensaje.
    printf("%s", $error -> getMessage())
}finally{
    // ...
}

// --------------------------------- //
// ------ Crear propio tipado ------ //
// --------------------------------- //

# Nuestra clase de Expecion personalizada.
class NoDivideException extends Exception{};

try
{
    // Se lanza una excepcion de tipo (NoDivideException).
    if(10 / 2 !== 5) return new NoDivideException("Mensaje");
}
catch(NoDivideException $noDivisible){
    // Se recibe la excepcion.

    $noDivisible -> getMessage();
}

// ---------------------------------------- //
// ------ Recibir varias excepciones ------ //
// ---------------------------------------- //

try{

    /* Se envian las excepciones. */

    if(){
        throw "Mensaje";
    }

    if(){
        return new Exception("Mensaje")
    }

    if(){
        return new NoDivideException("Mensaje");
    }

/**
 * PHP permite recibir numerosas excepciones.
 */
}catch(Throwable $throw){
    // ...
}catch(Exception $exception){
    // ...
}catch(NoDivideException $noDivisible){
    // ...
}

File: 1 - BackEnd/Vanilla/Python/estructuras-de-control/try-catch.py
# ##########=====================================##########
# ######===--- Try Except y lanzar excepciones ---===######
# ##########=====================================##########

# ------------------------------ #
# ------ Excepcion normal ------ #
# ------------------------------ #

try:
    # La palabra reservada (raise) permite lanzar excepciones.
    raise Exception("Variable: $hola")

except Exception as error:
    # El error se recibe.
    print(f"Error: {error}")

finally:
    # Bloque final de ejecución.
    print("Bloque finally ejecutado.")

# -------------------------------- #
# ------ Objeto (Exception) ------ #
# -------------------------------- #

try:
    # Se puede crear una instancia (Exception) y lanzarla.
    raise Exception("Mensaje")
except Exception as error:
    # El objeto lanzado posee la propiedad (args) que devuelve los errores en una (tupla).
    error.args[0]

# --------------------------------- #
# ------ Crear propio tipado ------ #
# --------------------------------- #

# Nuestra clase de Expecion personalizada.
class NoDivideException(Exception):
    pass

try:
    # Se lanza una excepción de tipo (NoDivideException).
    if 10 / 2 != 5:
        raise NoDivideException("Mensaje")

except NoDivideException as no_divisible:
    # Se recibe la excepción.
    no_divisible.args[0]

# ---------------------------------------- #
# ------ Recibir varias excepciones ------ #
# ---------------------------------------- #

try:
    # Se envían las excepciones.
    
    if True:  # Cambiar a la condición real
        raise Exception("Mensaje")

    if True:  # Cambiar a la condición real
        raise Exception("Mensaje")
    
    if True:  # Cambiar a la condición real
        raise NoDivideException("Mensaje")

except NoDivideException as no_divisible:
    no_divisible.args[0]
except Exception as exception:
    exception.args[0]